filament/src/materials/dof/dof.mat

```diff
@@ -477,6 +477,7 @@ void fastTile(inout vec4 color, inout float alpha,
     }
     alpha = cocToAlpha(kernelSize);
     color *= rcp(sampleCount(ringCountFast));
+    color = saturate(color);
```

Comment:
Doesn't the DoF work on HDR values? If so, `saturate()` is the wrong thing to do. `saturateMediump()` might be what you want though. Also, is it because values go too high/to infinity or because we have `NaN`s?

Reply:
Based on the screenshots in the bug, it seems like what's needed is a `max(0.0f, color)` instead?

Reply:
`saturateMediump()` didn't work.
![image](https://github.com/google/filament/assets/27909137/a571bce0-edc8-40d9-9a8f-d587433c42f5)
![image](https://github.com/google/filament/assets/27909137/bd86014c-fafe-42bd-a876-6ac412d7cb21)

Also, `max(0.0f, color)` didn't work.
![image](https://github.com/google/filament/assets/27909137/4f71fd49-8cb9-49b4-be10-89c9b9fca9fe)

And, I think it's because values go too high/to infinity because `min(1.0f, color);` works.

Reply:
Ah right you are on desktop so `saturateMediump()` is a no-op. Again, `min(1.0, color)` is wrong if the data is in HDR, which it may not be if DoF works on pre-tonemapped data.

Reply:
I found implementation of `saturateMediump()` [here](https://gist.github.com/romainguy/a2e9208f14cae37c579448be99f78f25?permalink_comment_id=2194308#gistcomment-2194308).  
And `min(65504.0f, color)` looks working for me. Is it okay to replace `saturate(color)` with `min(65004.0f, color)`?

Reply:
And does it imply we can use `saturateMediump()` for all device not only mobile?